shared/ssa/codeql/ssa/Ssa.qll

```diff
@@ -359,13 +359,17 @@ module Make<LocationSig Location, InputSig<Location> Input> {
      * either a read (when `k` is `SsaActualRead()`) or an SSA definition (when
      * `k` is `SsaDef()`).
      *
-     * Unlike `Liveness::ref`, this includes `phi` nodes.
+     * Unlike `Liveness::ref`, this includes `phi` nodes and pseudo-reads
+     * associated with uncertain writes.
      */
     pragma[nomagic]
     predicate ssaRef(BasicBlock bb, int i, SourceVariable v, SsaRefKind k) {
       variableRead(bb, i, v, _) and
       k = SsaActualRead()
       or
+      variableWrite(bb, i, v, false) and
+      k = SsaActualRead()
```

Comment:
Why is it that we have "pseudo-reads associated with uncertain writes"?

Reply:
That makes it much simpler to calculate `uncertainWriteDefinitionInput`.